src/BotPlutusInterface/ExBudget.hs

```diff
@@ -46,11 +46,18 @@ estimateBudget pabConf txFile = do
   let txBudget = do
         body <- txBody
         budget <- budgetRes
-        (spendingBudgets, policyBudgets) <- mkBudgetMaps budget body
+        let scaledBudget = fmap (fmap $ scaleBudget pabConf.pcBudgetMultiplier) budget
+        (spendingBudgets, policyBudgets) <- mkBudgetMaps scaledBudget body
         Right $ TxBudget spendingBudgets policyBudgets
 
   return txBudget
 
+-- | Scale the budget by the multipliers in config
+scaleBudget :: Rational -> CAPI.ExecutionUnits -> CAPI.ExecutionUnits
+scaleBudget scaler (CAPI.ExecutionUnits steps mem) = CAPI.ExecutionUnits (scale steps) (scale mem)
```

Comment:
I was thinking - maybe scaling should be capped by maximum CPU and MEM values from protocol parameters?

We don't know exactly how far budget need to be scaled. Lets say we scale all budgets x1.2. If calculated, say, memory budget is 14_000_000, scaling x1.2 will give 16 800 000, which will push it over the maximum budget for the testnet which is 16_000_000, but maybe that transaction could fit into 16_000_000. And there is no point to put budget bigger than maximum anyway.

Reply:
Actually yeah I completely agree, it'd outright break if we got within 20% of max for the real value - I'll add the cap